Doc updates including new page for troubleshooting

bcantoni opened the pull request:
This PR includes a batch of doc updates including:
- add new troubleshooting page showing how to capture REST API traffic
- rewrite sign in / sign out page to be more clear, in particular server vs online, and to - mention switch_site
- dev-guide, index, samples, versions: some general edits and improvements
- api-ref: added auth.switch_site (from #655)

Nothing in here is tied to any code changes, so it can be merged into gh-pages when ready. I'll squash all the commits before merging.

bcantoni commented:
These pages are reviewable - one more page coming.

shinchris commented on docs/api-ref.md:
```diff
@@ -136,18 +133,35 @@ REST API: [Sign Out](https://help.tableau.com/current/api/rest_api/en-us/REST/re
-[Server](#server)
+**Parameters**
+
+`site_id` | This corresponds to the `contentUrl` attribute in the Tableau REST API. The `site_id` is the portion of the URL that follows the `/site/` in the URL. For example, "MarketingTeam" is the `site_id` in the following URL *MyServer*/#/site/**MarketingTeam**/projects.
```
This method takes in a site_item, which contains the contentUrl field.

bcantoni replied:
Whoops, totally got that wrong. Fixing now with a better example too.

shinchris commented on docs/index.md:
```diff
@@ -49,13 +48,15 @@ development version are subject to change at any time before the next stable rel
-To install TSC onto a machine without internet connection, use the following steps:
+To install TSC onto a machine without an internet connection, use the following steps:
+
+1) Ensure that Python 3.6 or higher is installed.
```
I think we should say 3.5 or higher here as well for consistency

bcantoni replied:
Ack

shinchris commented on docs/sign-in-out.md:
```diff
@@ -3,46 +3,117 @@ title: Sign In and Out
+import tableauserverclient as TSC
+
+tableau_auth = TSC.PersonalAccessTokenAuth('TOKEN_NAME', 'TOKEN_VALUE', 'SITENAME')
+server = TSC.Server('https://SERVER_URL')
```
I think it would be helpful to add `use_server_version=True` here or `server.use_server_version()` below, to exemplify best practices

bcantoni replied:
Agreed

shinchris reviewed (changes requested):
Lots of really useful additions!

shinchris reviewed (approved):
Looks good!

bcantoni commented:
Thanks Chris for the feedback - I squashed and rebased so it can merge as one commit. (And I did a trim all trailing white space on api-ref.md to clean that up.)
